auth/feature-tests/component/token.generation.test.ts

```diff
@@ -0,0 +1,171 @@
+
+// --- Step 1: Initiate User Registration (Sign Up) ---
+async function signUpUser(username: string, password: string, email: string): Promise<void> {
+  const signUpCommand = new SignUpCommand({
+    ClientId: clientId,
+    Username: username,
+    Password: password,
+    UserAttributes: [
+      {
+        Name: 'email',
+        Value: email,
+      },
+    ],
+  });
+
+  try {
+    const signUpResponse = await cognitoClient.send(signUpCommand);
+    console.log('Sign-up successful:', signUpResponse);
+    // You might need to handle user confirmation here (e.g., via a confirmation code sent to their email)
+  } catch (error) {
+    console.error('Error signing up:', error);
+    throw error;
+  }
+}
+
+// --- Step 2: Initiate Authentication (Get Authorization Code) ---
+async function initiateAuthPKCE(): Promise<{ authorizationCode: string; codeVerifier: string }> {
+  const codeVerifier = generateCodeVerifier();
+  const codeChallenge = generateCodeChallenge(codeVerifier);
+  const codeChallengeMethod = 'S256';
+
+  // Construct the authorization URL
+  const authorizationUrl =
+    `https://${userPoolId.split('_')[0]}.auth.${region}.amazoncognito.com/oauth2/authorize` +
+    `?client_id=${clientId}` +
+    `&response_type=code` +
+    `&scope=openid` // Add the scopes your application needs
+    + `&redirect_uri=${encodeURIComponent(redirectUri)}` +
+    `&code_challenge=${codeChallenge}` +
+    `&code_challenge_method=${codeChallengeMethod}`;
+
+  console.log('Redirect the user to this URL for authorization:', authorizationUrl);
+
+  // In a real web application, you would redirect the user to this URL.
+  // After the user authenticates with Cognito, they will be redirected back to your `redirectUri`
+  // with an authorization code in the query parameters (e.g., `your_redirect_uri?code=YOUR_AUTH_CODE`).
+
+  // For this example, we'll simulate getting the authorization code.
+  const simulatedAuthCode = 'YOUR_SIMULATED_AUTHORIZATION_CODE'; // Replace with the actual code from the redirect
+  return { authorizationCode: simulatedAuthCode, codeVerifier };
+}
+
+// --- Step 3: Exchange Authorization Code for Tokens ---
+async function exchangeCodeForTokens(authorizationCode: string, codeVerifier: string): Promise<{ idToken: string; accessToken: string; refreshToken: string }> {
+  const initiateAuthCommand = new InitiateAuthCommand({
+    ClientId: clientId,
+    AuthFlow: 'USER_SRP_AUTH',
+    AuthParameters: {
+      'GRANT_TYPE': 'authorization_code',
+      'CODE': authorizationCode,
+      'REDIRECT_URI': redirectUri,
+      'CODE_VERIFIER': codeVerifier
+    },
+  });
+
+  try {
+    const authResponse = await cognitoClient.send(initiateAuthCommand);
+    console.log('Token exchange successful:', authResponse);
+    return {
+      idToken: authResponse.AuthenticationResult?.IdToken!,
+      accessToken: authResponse.AuthenticationResult?.AccessToken!,
+      refreshToken: authResponse.AuthenticationResult?.RefreshToken!,
+    };
```

Comment:
FYI - I tried this previously, however, this doesn't simulate how the app client uses us since it uses `initiateAuth`